plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/DatabaseTasksTree.java

```diff
@@ -514,9 +519,15 @@ private void addCheckedItem(TreeItem item, List<DBTTask> tasks) {
             if (parentElement instanceof DBPProject) {
                 DBPProject project = (DBPProject) parentElement;
                 // First add all tasks folders belonging to this project
-                children.addAll(allTasksFolders.stream().filter(taskFolder -> taskFolder.getProject() == parentElement).collect(Collectors.toList()));
+                children.addAll(allTasksFolders.stream()
+                    .filter(taskFolder -> taskFolder.getProject() == parentElement)
+                    .sorted((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()))
+                    .collect(Collectors.toList()));
                 // Then check all tasks belonging to this project without folder
-                List<DBTTask> thisProjectTasksWithoutFolder = allTasks.stream().filter(task -> task.getTaskFolder() == null && task.getProject() == parentElement).collect(Collectors.toList());
+                List<DBTTask> thisProjectTasksWithoutFolder = allTasks.stream()
+                    .filter(task -> task.getTaskFolder() == null && task.getProject() == parentElement)
+                    .sorted((o1, o2) -> o1.getName().compareToIgnoreCase(o2.getName()))
+                    .collect(Collectors.toList());
```

Comment:
Is sorting here essential? We can define a custom comparator for the viewer instead.